packages/common/container-definitions/src/loader.ts

```diff
@@ -245,6 +245,11 @@ export interface IContainerEvents extends IEvent {
 	 * @see {@link IContainer.isDirty}
 	 */
 	(event: "saved", listener: (dirty: boolean) => void);
+
+	/**
+	 * Emitted when the some of the properties related to the container are initialized or updated.
+	 */
+	(event: "metadataUpdate", listener: (metadata?: Record<string, unknown> | undefined) => void);
```

Comment:
Why can it be undefined? I'd rather see {} used in such case (if it implies - all properties are empty), not undefined.
Please add comments about semantics how it should be interpreted by consumer. I see two ways:
1. **Merging properties / communicates only what changed**: If some property is missing on this payload, but existed before (in `containerMetadata` state below), then it did not change and stays the same. If property needs to be deleted, it's value will be set to undefined in `metadata` value on this event.
2. **Overwrite**: If some property is missing in `metadata`, it means it is removed from set.

Reply:
The join session will always return the labels and not just the updates even if they are empty. So as far as labels is concerned, we will have the state always. I want to go with this semantic for these updates as you previously mentioned:

Merging properties / communicates only what changed: If some property is missing on this payload, but existed before (in containerMetadata state below), then it did not change and stays the same. If property needs to be deleted, it's value will be set to undefined in metadata value on this event.

As then the update provider does not need to bother about all props and even container can itself add props to the metadata as the service might not know all the props, so complete overwrite semantics does not work. If someone wants to reset a prop, then they need to provide that in update otherwise it will remain same.